### PR TITLE
DDCE-4736:ERS-CSOP-EMI_OptionsRCL_V4

### PR DESCRIPTION
**Pull request by harish-korada1**

### CODE REVIEW

File: test/services/validation/ERSValidationConfigCSOP.scala [hunk 1]
```diff
@@ -116,8 +118,24 @@ class CSOPOptionsRCLTest extends PlaySpec with ERSValidationCSOPRCLTestData with
       )
     }
 
-  }
+    "when colB is answered yes then remaining columns from C to I are mandatory and if entered invalid or missing data then it should throw error " in {
+      val row = Row(1,getWronglyEnteredCellData)
+      val resOpt: Option[List[ValidationError]] = validator.validateRow(row)
+      resOpt.withErrorsFromMessages.get.size mustBe 3
```
Comment (MRBeresford) [inline test/services/validation/ERSValidationConfigCSOP.scala:124]: Is it worth checking what the errors are? similar to the test above `"when colB is answered yes, colC is a mandatory field" in {...`

File: test/services/validation/CSOPTestData/ERSValidationCSOPRCLTestData.scala [hunk 2]
```diff
@@ -80,22 +80,22 @@ trait ERSValidationCSOPRCLTestData {
...
       Cell("G", rowNumber, "AB123456"),
-      Cell("G", rowNumber, ""),
+      Cell("G", rowNumber, "AB123456"),
       Cell("H", rowNumber, "123/XZ55555555"),
       Cell("H", rowNumber, "1234/12345/67890abcd"),
-      Cell("H", rowNumber, ""),
+      Cell("H", rowNumber, "1234/12345/67890abcd"),
       Cell("I", rowNumber, "Yes"),
-      Cell("I", rowNumber, " "),
+      Cell("I", rowNumber, "YyEeSs"),
```
Comment (MRBeresford) [inline test/services/validation/CSOPTestData/ERSValidationCSOPRCLTestData.scala:98]: I don't think the changes from `" "` and  `""` are necessary, in this example we now just have two rows with `YyEeSs`?

**Review by MRBeresford — CHANGES_REQUESTED** [review]
Some comments

File: test/services/validation/CSOPTestData/ERSValidationCSOPRCLTestData.scala (hunk 2 above)
Comment (MRBeresford) [reply test/services/validation/CSOPTestData/ERSValidationCSOPRCLTestData.scala]: @harish-korada1 I still think you need the test case with an empty string for example, I dont think we should just delete them.

**Review by MRBeresford — APPROVED** [review]
LGTM
